tests/unit/test_charm.py

```diff
@@ -250,13 +250,13 @@ def test__on_loki_push_api_endpoint_joined(
     @patch.object(Container, "pull", new=pull_empty_fake_file)
     @patch.object(Container, "restart")
     @patch.object(Container, "push")
-    def test__on_loki_push_api_endpoint_departed(
+    def test__on_loki_push_api_endpoint_departed_0_endpoints(
         self, mock_push: MagicMock, mock_restart: MagicMock
     ):
         """Test Loki config is not in config file when LokiPushApiEndpointDeparted is fired."""
         mock_restart.restart.return_value = True
         self.harness.charm._loki_consumer = Mock()
-        self.harness.charm._loki_consumer.loki_push_api = "http://loki:3100:/loki/api/v1/push"
+        self.harness.charm._loki_consumer.loki_endpoints = []
```

Comment:
I don't think we should use mocks for these tests, it makes me very uncomfortable

Reply:
For evidence, these mocks have hidden this bug all along https://github.com/canonical/loki-k8s-operator/pull/138

Reply:
We will address the removal of all these mocks in https://github.com/canonical/grafana-agent-k8s-operator/issues/45